tomSawkins/typed-get-prop
# Initial commit

### Summary
Initial commit containing your original code (under `src/index.ts`) with various supporting scripts.

#### Features
- Script to generate function overloads with as many nested parameters as needed. Default is 20.
- Readme contains example usage, summary, value proposition and status badges for [npm](https://npmjs.com), [travis-ci](https://travis-ci.org), [coveralls](https://coveralls.io/) and [greenkeeper](https://greenkeeper.io). You will need to create accounts with these services if you don't already. Because this is an open-source project the free-tier in each service is fine. I can help with configuration if you run into any issues.

### Releasing
I've got into a bit of a pattern with creating new libraries now where I just copy the last library I published to quickly support:
- Linting
- Building
- Tests
- Conventional commits (for easier semantic versioning and change-log)
- Packaging for deployment
- Releasing and shipping

The usual workflow for contributing as a primary author would go as follows:

1. Make changes (to `./support/generate.ts` and `./support/template.ts`)

2. Run `npm run commit` to get the prompt to enter your commit message. Alternatively, you could write it yourself using `git commit -m "blah"` and the husky `commit-msg` hook should ensure your message is formatted correctly.

3. Run `npm run release`. This will build the app, lint it, test it, package everything under dist, version bump, update the change log, commit in, push your commits up to origin and copy the latest package version to dist (post version bump).

4. Run `npm run ship`. Do this when you're ready to publish the latest version to npm.
